### DISCUSSION TITLE
Use FluentAssertions outside of tests

### DISCUSSION THREAD

**Discussion by reager** [fluentassertions/fluentassertions#2795] [General]
I like using FluentAssertions in my guard clauses instead of if/throw. 

Is there any downside to this? Any reason for not doing this at all?

Searched but didnt find any relevant information.

**Comment by dennisdoomen** [comment]
It was not designed for that, so we cannot guarantee it will work in all scenarios. We do care about performance, so if you run into anything, there _is_ a chance that we'll fix it after all.
